src/toplevel.c

```diff
@@ -585,8 +592,13 @@ jl_value_t *jl_parse_eval_all(const char *fname, size_t len)
             jl_rethrow();
         }
         else {
-            jl_rethrow_other(jl_new_struct(jl_loaderror_type, fn, ln,
-                                           jl_exception_in_transit));
+            if(jl_typeis(jl_exception_in_transit, jl_initerror_type)) {
```

Comment:
I don't think this makes sense. Whether an error causes loading a file to fail is orthogonal to the underlying cause of the error. For example if this is "x.jl":

```
using Foo  # throws InitError

Foo.f()
```

we won't get a LoadError for loading x.jl.